### PR TITLE
added Verifier and Semver

### PR DESCRIPTION
**Pull request by wstrametz**
added Verifier and Semver

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -2,16 +2,33 @@ name: Docker Build
 
 on:
   push:
-    branches: [ buildfile ]
+    branches: [ verifier ]
```
Comment (dastansam) [inline .github/workflows/docker-build.yml:5]:
```suggestion
    branches: [ main ]
    
pull:
    branches: [ main ]
```

Comment (dastansam) [reply .github/workflows/docker-build.yml:5]: will check when push on main and when PR opens to `main`

File: verifier/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        DEFAULT_PROOF_JSON
+    };
+
+    let receipt_json: Vec<u8> = fs::read(proof_json_path)
+        .unwrap_or_else(|_| panic!("Failed to read file at {}", proof_json_path));
+
+    let receipt: Receipt =
+        serde_json::from_slice(&receipt_json).expect("Failed to parse proof JSON");
+    let result_string =
+        String::from_utf8(receipt.journal.bytes).expect("Failed to convert bytes to string");
```
Comment (dastansam) [inline verifier/src/main.rs:28]: doesn't call `verify`

File: docs/guest-hyperfridge.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
+# Guest progroam (hyperfridge)
+
+## Overview
+
+This module provides the logic to check the soundness of data coming from the banking backend - in Risc-Zero terminology the [guest][guest] code of hyperfridge. Risc-Zero framework creates a [seal][seal] around a specific compilation unit and version. Executingthis binary is used to create a [stark] (captured in a JSON) which can be validated automatically in any Rust module (e.g. on Polkadot) but also on EVM based ledgers (using [eth-verifier-contract]) - but the usage is not limited to blockchain technologies.
+
+The proof aims for following properties:
+
+- **Proofing soundness**: We want to prove that the presented data is correct and relyable and has not been tempered with.
```
Comment (dastansam) [inline docs/guest-hyperfridge.md:9]:
```suggestion
- **Proofing soundness**: We want to prove that the presented data is correct and reliable and has not been tampered with.
```

File: docs/guest-hyperfridge.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+We achieve this by checking signatures in the data which are provided by multiple entities. No entity alone should be able
+to generate (or fake) the proof on its own. So either the bank or the banks client is able to generate fake proofs,
+because you would need the private keys of both parties. The EBICS protocol (Electronic Banking Internet Communication Standard)
+defines the key ceremony which has been used for many years.
+- **Privacy**: Financial data - like medical data - is prone to highest data security standards. Bank documents contain names
+and bank details of clients, which should not be leaked. With Zerko-Knowledge technology we are able generate proofs that.
```
Comment (dastansam) [inline docs/guest-hyperfridge.md:15]:
```suggestion
and bank details of clients, which should not be leaked. With Zero-Knowledge technology we are able generate proofs that,
```

File: verifier/src/main.rs (hunk 2 above)
Comment (wstrametz) [reply verifier/src/main.rs]: done

**Review by vladae36 — APPROVED** [review]

**Pull request merged**